sync/class.jetpack-sync-actions.php

```diff
@@ -340,3 +355,5 @@ static function init_sync_cron_jobs() {
 
 // We need to define this here so that it's hooked before `updating_jetpack_version` is called
 add_action( 'updating_jetpack_version', array( 'Jetpack_Sync_Actions', 'do_initial_sync' ), 10, 2 );
+
+add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'setup_woocommerce' ) );
```

Comment:
plugins_loaded is before init, and is the earliest we can detect whether woo has been enabled. Performance-wise, I don't think this adds much overhead. Which part do you think is expensive?